### PR TITLE
FRW-19 Support Symfony packages v6.

### PR DESCRIPTION
**Pull request by geega**
- Developer(s): @geega

- Ticket: https://spryker.atlassian.net/browse/FRW-19

- Release Group: https://release.spryker.com/release-groups/view/4549

- merge: merge

#### Release Table

   Module                | Release Type         | Constraint Updates         |
   :--------------------- | :------------------------ | :--------------------- |
   Silexphp              | patch (currently 0.x)  |                       |

-----------------------------------------

##### Change log

Fixes

- Adjusted `\Silex\Application`, `\Silex\AppArgumentValueResolver`, `\Silex\WebTestCase` to be compatible with `symfony/http-kernel:^5.0.0 || ^6.0.0`.
- Adjusted `\Silex\Provider\Routing\RedirectableUrlMatcher`, `\Silex\RedirectableUrlMatcher`, `\Silex\Provider\Routing\LazyRequestMatcher`, `\Silex\Provider\Routing\RedirectableUrlMatcher`, `\Silex\RedirectableUrlMatcher` to be compatible with `symfony/routing:^5.0.0 || ^6.0.0`.

Improvement

-  Increased minimum PHP version to 8.
- Adjusted `\Silex\CallbackResolver`,  `\Silex\ConstraintValidatorFactory`, `Silex\Provider\Twig\RuntimeLoader` for compatibility with `spryker/container`.
- Dropped impossible `symfony/event-dispatcher:^3.0.0 || ^4.0.0` constraints due to EOL PHP version incompatibility.
- Dropped impossible `symfony/http-kernel:^4.3.0` constraints due to EOL PHP version incompatibility.
- Dropped impossible `symfony/routing:^3.0.0 || ^4.0.0` constraints due to EOL PHP version incompatibility.
- Adjusted `composer.json` to allow for new `symfony/event-dispatche:^6.0.0` major.
- Adjusted `composer.json` to allow for new `symfony/http-kernel:^6.0.0` major.
- Adjusted `composer.json` to allow for new `symfony/routing:^6.0.0` major.

Deprecation

- Deprecated `Silex\Application\FormTrait`.

### CODE REVIEW

**Review by dimitriyTsemma — APPROVED** [review]

**Review by fabiospryker — APPROVED** [review]

File: src/Silex/ApplicationTrait.php [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+namespace Silex;
+
+use Composer\Semver\VersionParser;
+use Composer\InstalledVersions;
+use Symfony\Component\HttpFoundation\Request;
+use Symfony\Component\HttpFoundation\Response;
+use Symfony\Component\HttpKernel\HttpKernelInterface;
+
+
+if (InstalledVersions::satisfies(new VersionParser, 'symfony/http-kernel', '^6.0.0')) {
```
Comment (asmarovydlo) [inline src/Silex/ApplicationTrait.php:12]: Do we still support `spryker/silexphp`? @stereomon

Comment (stereomon) [reply src/Silex/ApplicationTrait.php:12]: We shouldn't

Comment (asmarovydlo) [reply src/Silex/ApplicationTrait.php:12]: @geega does this package update block us from making the suite-nonsplit Travis green?

Comment (geega) [reply src/Silex/ApplicationTrait.php:12]: @asmarovydlo  Yes. Only for fixing Travis test issues.

File: src/Silex/Provider/Routing/RedirectableUrlMatcher.php [hunk 2]
```diff
@@ -24,7 +24,7 @@ class RedirectableUrlMatcher extends BaseRedirectableUrlMatcher
     /**
      * {@inheritdoc}
      */
-    public function redirect($path, $route, $scheme = null)
+    public function redirect(string $path, string $route, string $scheme = null): array
```
Comment (asmarovydlo) [inline src/Silex/Provider/Routing/RedirectableUrlMatcher.php:27]: It won't work for an older version of `symfony/routing`, please fix it the same way as for other cases.

File: src/Silex/ApplicationTrait.php (hunk 1 above)
Comment (asmarovydlo) [inline src/Silex/ApplicationTrait.php:12]: Please use `Kernel::MAJOR_VERSION` instead, for performance reasons.

**Review by asmarovydlo — CHANGES_REQUESTED** [review]
Please add:
1. Release table
2. Changelogs
3. Labels to the PR.

File: src/Silex/RedirectableUrlMatcherTrait.php [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+    {
+        /**
+         * {@inheritdoc}
+         */
+        public function redirect(string $path, string $route, string $scheme = null): array
+        {
+            return $this->executeRedirect($path, $route, $scheme);
+        }
+
+
```
Comment (asmarovydlo) [inline src/Silex/RedirectableUrlMatcherTrait.php:47]: Please remove the empty line.

File: src/Silex/RedirectableUrlMatcherTrait.php [hunk 4]
```diff
@@ -0,0 +1,58 @@
...
+         * @return array
+         */
+        abstract protected function executeRedirect(string $path, string $route, string $scheme = null): array;
+    }
+} else {
+    trait RedirectableUrlMatcherTrait
+    {
+        /**
+         * {@inheritdoc}
+         */
```
Comment (asmarovydlo) [inline src/Silex/RedirectableUrlMatcherTrait.php:41]: The return type and parameters declaration missing. (Several similar places in the PR)

File: src/Silex/RedirectableUrlMatcher.php [hunk 5]
```diff
@@ -11,6 +11,8 @@
 
 namespace Silex;
 
+use Composer\InstalledVersions;
+use Composer\Semver\VersionParser;
```
Comment (asmarovydlo) [inline src/Silex/RedirectableUrlMatcher.php:15]: Not used

**Review by asmarovydlo — CHANGES_REQUESTED** [review]
Let's follow the conventions https://spryker.atlassian.net/wiki/spaces/CORE/pages/523730976/Release+Notes+and+Changelogs

Allow the next major of symfony/routing (^6.0). => 
- Adjusted `composer.json` to allow for new `symfony/routing:^6.0.0` major. 
Please adjust for **ALL** the PRs in the RG.

File: src/Silex/ApplicationTrait/ApplicationTrait.symfony5.php [hunk 6]
```diff
@@ -0,0 +1,48 @@
...
+    }
+
+    /**
+     * @param \Symfony\Component\HttpFoundation\Request $request
+     * @param int $type
+     * @param bool $catch
+     *
+     * @return \Symfony\Component\HttpFoundation\Response
+     */
+    abstract protected function executeHandle(Request $request, int $type = HttpKernelInterface::MASTER_REQUEST, bool $catch = true): Response;
```
Comment (asmarovydlo) [inline src/Silex/ApplicationTrait/ApplicationTrait.symfony5.php:47]: Is is already defined in ApplicationTraitCommon

**Review by asmarovydlo — CHANGES_REQUESTED** [review]

File: src/Silex/Application.php [hunk 7]
```diff
@@ -605,8 +605,14 @@ public function run(?Request $request = null)
...
      * terminate() method yourself if you want the finish filters to be run.
+     *
+     * @param \Symfony\Component\HttpFoundation\Request $request
+     * @param int $type
+     * @param bool $catch
+     *
+     * @return \Symfony\Component\HttpFoundation\Response
      */
-    public function handle(Request $request, $type = HttpKernelInterface::MASTER_REQUEST, $catch = true)
+    public function handle(Request $request, int $type = HttpKernelInterface::MASTER_REQUEST, bool $catch = true): Response
```
Comment (asmarovydlo) [inline src/Silex/Application.php:615]: Please Travis issues, they are reated

Comment (asmarovydlo) [inline src/Silex/Application.php:615]: It won't work for `symfony/http-kernel` < 5
See https://github.com/symfony/http-kernel/blob/v4.4.37/HttpKernelInterface.php

The same issue with routing-related changes. Please verify.

**Review by asmarovydlo — CHANGES_REQUESTED** [review]

File: composer.json [hunk 8]
```diff
@@ -17,39 +17,41 @@
         }
     ],
     "require": {
```
Comment (asmarovydlo) [inline composer.json:19]: All composer changes should be mentioned in the changelogs.

File: .github/workflows/ci.yml [hunk 9]
```diff
@@ -59,7 +58,7 @@ jobs:
       fail-fast: false
       matrix:
         php-version: [
-            '7.3'
+            '8.1'
```
Comment (asmarovydlo) [inline .github/workflows/ci.yml:61]: Prefer lowest should work with php `8.0`

**Review by asmarovydlo — CHANGES_REQUESTED** [review]

**Review by asmarovydlo — APPROVED** [review]

File: .github/workflows/ci.yml [hunk 10]
```diff
@@ -12,8 +12,7 @@ jobs:
       fail-fast: false
       matrix:
         php-version: [
-            '7.4',
-            'nighly'
+            '8.1'
```
Comment (dereuromark) [inline .github/workflows/ci.yml:15]: This should be 8.0, 8.2
no?

**Pull request merged**